### PR TITLE
feat: [DHIS2-15187][DHIS2-15190] Working list bulk actions

### PR DESCRIPTION
**Pull request by eirikhaugstulen**
Tech-summary:
- Added a selection checkbox in the working list
- Added a generic BulkActionsBar component

Event bulk actions:
- Complete events
- Delete events
- Deselect

Tracked entity actions:
- Complete enrollments
    - Does a dryRun to the api to look for errors
    - Partial update if any invalid data
    - Deselects partial imported enrollments if any
- Delete enrollments
    - Able to delete active, completed and cancelled enrollments (user selects combination)
- Deselect 

Left to do:
- [x] Tests for tracked entity lists
- [x] Documentation

### CODE REVIEW

File: src/core_modules/capture-core/components/WorkingLists/EventWorkingListsCommon/EventBulkActions/Actions/CompleteAction/hooks/useBulkCompleteEvents.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                // $FlowFixMe - business logic dictates that events are not undefined
+                events: events.activeEvents.map(event => ({
+                    ...event,
+                    status: 'COMPLETED',
+                })),
+            },
+        }),
+        {
+            onError: () => {
+                showAlert({ message: i18n.t('An error occurred while completing events') });
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WorkingLists/EventWorkingListsCommon/EventBulkActions/Actions/CompleteAction/hooks/useBulkCompleteEvents.js:76]: There are cases where only some of the selected events can be completed (e.g. some events are missing values ​​for required data elements). Does it make sense to add logic similar to `hasPartiallyUploadedEnrollments` with more detailed errors in these cases? Thanks!

**Review by simonadomnisoru — COMMENTED** [review]
Good job on this feature, the test coverage is quite impressive 👏 !

I've left a few comments in the code. Additionally, I have some questions about missing functionality:
1. Ask user to complete program when stage is complete - (`remindCompleted` in program stage's metadata)
2. Ask user to create new event when stage is complete - (`allowGenerateNextVisit` in program stage's metadata)
3. Missing data access write checks. Besides verifying authority, shouldn't we also check if the user has the appropriate data write permissions for each specific case (tracked entity type, program, program stage)?

Will these issues be addressed in a future iteration of WL bulk actions, or should they be tackled in this PR? 
Thank you!

File: src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/TrackedEntityBulkActions/Actions/DeleteEnrollmentsAction/EnrollmentDeleteModal/EnrollmentDeleteModal.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        );
+    }
+
+    return (
+        <Modal
+            onClose={() => setIsDeleteDialogOpen(false)}
+            dataTest={'bulk-delete-enrollments-dialog'}
+        >
+            <ModalTitle>
+                {i18n.t('Delete selected enrollments')}
```
[the remark this answers is not in the document]
Comment (eirikhaugstulen) [reply src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/TrackedEntityBulkActions/Actions/DeleteEnrollmentsAction/EnrollmentDeleteModal/EnrollmentDeleteModal.js:118]: I agree that the current text is a bit confusing. I also think that using the `TETname` can lead to some unexpected texts. For example, `Lab samples's enrollments`. What about just using "Delete enrollments" and "Complete enrollments"?

File: src/core_modules/capture-core/components/WorkingLists/EventWorkingListsCommon/EventBulkActions/Actions/CompleteAction/hooks/useBulkCompleteEvents.js (hunk 1 above)
Comment (eirikhaugstulen) [reply src/core_modules/capture-core/components/WorkingLists/EventWorkingListsCommon/EventBulkActions/Actions/CompleteAction/hooks/useBulkCompleteEvents.js:79]: Yes, that makes sense! I've added some extra logic to handle these cases, so it should be good now!

File: src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/TrackedEntityBulkActions/Actions/DeleteTeiAction/DeleteTeiAction.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        onUpdateList,
+    });
+
+    if (!hasAuthority) {
+        return null;
+    }
+
+    return (
+        <>
+            <Button
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/TrackedEntityBulkActions/Actions/DeleteTeiAction/DeleteTeiAction.js:37]: The DeleteTeiaction action is missing the program access and trackedEntityType checks. Can something similar to how [WidgetProfile](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/WidgetProfile/WidgetProfile.component.js#L122-L124) currently works be implemented? Thanks!

File: src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/TrackedEntityBulkActions/Actions/CompleteAction/CompleteAction.js [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                    onChange={() => setCompleteEvents(prevState => !prevState)}
+                />
+
+            </div>
+        );
+    };
+
+    return (
+        <>
+            <Button
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/TrackedEntityBulkActions/Actions/CompleteAction/CompleteAction.js:173]: For both completing and deleting enrollments it is necessary to verify the program data write access. Can you add the checks with ConditionalTooltip? 🙏

**Comment by eirikhaugstulen** [comment]
Thanks, @simonadomnisoru! Should be implemented now 😊

**Review by simonadomnisoru — APPROVED** [review]
Good job! 🥳

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.42,2.41.3,2.40.6,2.39.8 versions

**Pull request merged**